app/controllers/application_controller.rb

```diff
@@ -1,2 +1,14 @@
 class ApplicationController < ActionController::Base
+  before_action :set_current_user
+
+  def require_user_logged_in!
+    # allows only logged in user
+    redirect_to sign_in_path, alert: 'You must be signed in' if Current.user.nil?
+  end
```

Comment:
Do you think this should also be a private method?

Reply:
ma'am In case if will use this method in other controller. That's why I haven't made it private !!

Reply:
So we can't access private methods in subclasses?